FlickrViewer/FlickrViewer/FlickrViewerForm.cs

```diff
@@ -10,118 +12,108 @@
-                  "Flickr Error", MessageBoxButtons.OK, 
-                  MessageBoxIcon.Error );
+                var result = MessageBox.Show(
+                    "Cancel the current Flickr search?",
+                    "Are you sure?", MessageBoxButtons.YesNo,
+                    MessageBoxIcon.Question);
+
+                // determine whether user wants to cancel prior search
+                if (result == DialogResult.No) return;
+                _flickrClient.CancelAsync();
+            }
+
+            // Flickr's web service URL for searches
+            var flickrUrl = string.Format(FlickrWebServiceUrlTemplate, Key, inputTextBox.Text.Replace(" ", ","));
+
+            imagesListBox.DataSource = null; // remove prior data source
             imagesListBox.Items.Clear(); // clear imagesListBox
-            imagesListBox.Items.Add( "Error occurred" );
-         } // end catch
-      } // end method searchButton_Click
-
-      // display selected image
-      private async void imagesListBox_SelectedIndexChanged(
-         object sender, EventArgs e )
-      {
-         if ( imagesListBox.SelectedItem != null )
-         {
-            string selectedURL =
-               ( ( FlickrResult ) imagesListBox.SelectedItem ).URL; 
-
-            // use WebClient to get selected image's bytes asynchronously
-            WebClient imageClient = new WebClient();
-            byte[] imageBytes = await imageClient.DownloadDataTaskAsync( 
-               __________________ );
-
-            // display downloaded image in pictureBox
-            MemoryStream memoryStream = new MemoryStream( imageBytes );
-            pictureBox.Image = Image.FromStream( memoryStream );
-         } // end if
-      } // end method imagesListBox_SelectedIndexChanged
-   } // end class FlickrViewerForm
-} // end namespace FlickrViewer
+            pictureBox.Image = null; // clear pictureBox
+            imagesListBox.Items.Add("Loading..."); // display Loading...
+
+            try
+            {
+                // invoke Flickr web service to search Flick with user's tags
+                _flickrTask = _flickrClient.DownloadStringTaskAsync(flickrUrl);
+
+                // await flickrTask then parse results with XDocument and LINQ
+                XDocument flickrXml = XDocument.Parse( await _flickrTask);
+
+                // gather information on all photos
+                var flickrPhotos =
+                    from photo in flickrXml.Descendants("photo")
+                    let id = photo.Attribute("id").Value
+                    let title = photo.Attribute("title").Value
+                    let secret = photo.Attribute("secret").Value
+                    let server = photo.Attribute("server").Value
+                    let farm = photo.Attribute("farm").Value
+                    select new FlickrResult
+                    {
+                        Url = string.Format(FlickrResultUrlTemplate, farm, server, id, secret)

```

Comment:
Missing the Title property. This causes the results list to have empty titles

Reply:
I would recommend always testing your code in the future. I know you have a Mac as well, so what I did was use VMWare, set up Windows 10, and then run the code on Visual Studio on the virtual machine.